plotFADC/FADCtest.nim

```diff
@@ -0,0 +1,180 @@
+import nimhdf5, os, ingrid/tos_helpers, sequtils, strutils
+import sets
+import plotly
+import tables
+import re
+import docopt
+
+
+when defined(linux):
+  const commitHash = staticExec("git rev-parse --short HEAD")
+  const currentDate = staticExec("date")
+else:
+  const commitHash = ""
+  const currentDate = ""
+
+const docTmpl = """
+InGrid/FADC reading and cuting.
+
+Usage:
+ FADCtest <HDF5file> [options]
+ FADCtest <HDF5file> --runNumber <runnum> [options]
+ FADCtest <HDF5file> --runNumber <runnum> --chipNumber <chipnum> [options]
+ FADCtest <HDF5file> --chipNumber <chipnum> [options]
+
+Options:
+ -h --help show this text
+ --runNumber      choose the run you would like to look at
+ --chipNumber     choose the chip you would like to look at
+ if not selected the program chooses the last values it gets from the file
+
+ """
+
+const doc = docTmpl % [commitHash, currentDate]
+
+proc readandcut*(h5f: var H5FileObj, runNumber: int, chip: int) =
+
+  ##get the ingrid data
+
+  var reco_group = rawDatabase() & $runNumber
+  var reco_group_chip = recoDataChipBase(runNumber) & $chip
+  var group = h5f[reco_group.grp_str]
+  var group_chip = h5f[reco_group_chip.grp_str]
+  #echo groups
+
+  var evNumbersDset = h5f[(group_chip.name / "eventNumber").dset_str]
+  let evNumbers = evNumbersDset[int64]
+
+  let chip_number = group_chip.attrs["chipNumber", int]
+  var pos_x_dset = h5f[(group_chip.name / "centerX").dset_str]
+  let pos_x = pos_x_dset[float64]
+  var eccentricity_dset = h5f[(group_chip.name / "eccentricity").dset_str]
+  let eccentricity = eccentricity_dset[float64]
+
+  ##get the fadc data
+
+  var fadc_eventNumber_name = eventNumberBasename(runNumber)
+  var fadc_evNumbersDset = h5f[fadc_eventNumber_name.dset_str]
+  let fadc_evNumbers = fadc_evNumbersDset[int64]
+  #echo fadc_evNumbers
+
+  var riseStart_name = riseStartBasename(runNumber)
+  var fadc_riseStartDset = h5f[riseStart_name.dset_str]
+  let fadc_riseStart = fadc_riseStartDset[uint16]
+
+  var fallStop_name = fallStopBasename(runNumber)
+  var fadc_fallStopDset = h5f[fallStop_name.dset_str]
+  let fadc_fallStop = fadc_fallStopDset[uint16]
+
+  var riseTime_name = riseTimeBasename(runNumber)
+  var fadc_riseTimeDset = h5f[riseTime_name.dset_str]
+  let fadc_riseTime = fadc_riseTimeDset[uint16]
+
+  var fallTime_name = fallTimeBasename(runNumber)
+  var fadc_fallTimeDset = h5f[fallTime_name.dset_str]
+  let fadc_fallTime = fadc_fallTimeDset[uint16]
+  #echo fadc_fallTime
+
+  ##process data and make cuts
+
+  let zipData = zip(evNumbers, eccentricity)
+  #echo zipData
+
+  let zipData_fadc = zip(fadc_evNumbers, fadc_fallTime)
```

Comment:
Clean up code above (only read the dataset actually used).
Might also want to use the new convenience interface from nimhdf5:
```nim
let fadc_fallTime = h5f[fallTime_name, uint16]  # yes, without conversion to dset_str
```
to read the data in one call, without an intermediate dataset, if you don't need that dataset anyways.